hooks/environment

```diff
@@ -12,6 +12,25 @@ check_command() {
   fi
 }
 
+# In `buildkite-agent` 3.66.0
+# (https://github.com/buildkite/agent/releases/tag/v3.66.0) the ability to
+# redact secrets (https://buildkite.com/docs/agent/v3/cli-redactor) was added.
+# However, because we may not always run that version of `buildkite-agent`, we
+# should conditionally enable it
+check_buildkite_agent_version_for_redaction() {
+  local version_output
+  version_output="$(buildkite-agent --version)"
+  local version
+  version=$(echo "$version_output" | grep -oP '\d+\.\d+\.\d+')
+  local min_version="3.66.0"
+  if [[ "$(printf '%s\n' "$min_version" "$version" | sort --version-sort | head -n1)" = "$min_version" ]]; then
```

Comment:
Still needs addressing

Either:

```suggestion
  if [[ "$(printf '%s\n%s' "$min_version" "$version" | sort --version-sort | head -n1)" = "$min_version" ]]; then
```

or:


```suggestion
  if [[ "$(echo -e "$min_version\n$version" | sort --version-sort | head -n1)" = "$min_version" ]]; then
```

Reply:
Same as above, does this add something extra or is it a matter of taste? Because it is functionong properly.

Reply:
changed in 8341afb07604de424ad9bd1e0880dc705022ef94